TestSupport/g6_diff_finder.py

```diff
@@ -0,0 +1,513 @@
+                with open(outfile_path, 'w') as graph_set_intersection_outfile:
+                    graph_set_intersection_outfile.write(
+                        json.dumps(
+                            graphs_in_first_and_second,
+                            indent = 4
+                        )
+                    )
+
+    def _get_infile_names(
+            self,
+            first_comparand_dict: dict,
+            second_comparand_dict: dict
+        ):
+        """
+        Uses pathlib.Path object's .parent attribute to get the directory of
+        the first .g6 input file, stored in the first_comparand_dict's
+        infile_path attribute. Then, gets the first_comparand_infile_name by
+        stripping the .g6 extension, and likewise for the
+        second_comparand_infile_name.
+
+        Args:
+            first_comparand_dict: Dict containing the key infile_name,
+                in addition to the key-value pairs of graph encodings mapped to
+                their first_line and the list of duplicate_line_nums
+            second_comparand_dict: Same structure as first_comparand_dict
+        
+        Returns:
+            first_comparand_infile_dir: Parent directory of the first .g6 file
+            first_comparand_infile_name: base name of the first .g6 file
+            second_comparand_infile_name: base name of the second .g6 file
+        
+        Raises:
+            G6DiffFinderException: If KeyError encountered when trying to
+                get value corresponding to key infile_path from either of the
+                two comparand dicts
+        """
+        try:
+            first_comparand_infile_path = \
+                Path(first_comparand_dict['infile_path']).resolve()
+        except KeyError as e:
+            raise G6DiffFinderException(
+                'Invalid dict structure: missing \'infile_path\'.') from e
+        else:
+            first_comparand_infile_dir = first_comparand_infile_path.parent
+            first_comparand_infile_name = \
+                first_comparand_infile_path.with_suffix('').name
+            try:
+                second_comparand_infile_name = \
+                    Path(
+                        second_comparand_dict['infile_path']
+                        ).with_suffix('').name
+            except KeyError as e:
+                raise G6DiffFinderException(
+                    'Invalid dict structure: missing \'infile_path\'.'
+                    ) from e
+            else:
+                return first_comparand_infile_dir, \
+                    first_comparand_infile_name, \
+                    second_comparand_infile_name
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        usage='python %(prog)s [options]',
+        description="""Tool to help interrogate and compare two .g6 files.
+
+- Determines if there are duplicates in the first and second comparand .g6 
```

Comment:
Could you tweak this message to indicate the filename pattern for the different output files, if any, so one can immediately know, from reading this description, which output files to expect and what to expect in them?